### DISCUSSION TITLE
View randomly doesn't update iOS 17

### DISCUSSION THREAD

**Discussion by arnauddorgans** [pointfreeco/swift-composable-architecture#2852] [Q&A]
### Description

I have a ScrollView containing a LazyVStack with four views, each with its scope.
Randomly, one of the views is not updated. 
When I add a tap gesture on the not updated view to print the state, the state prints correctly updated, but the update is not visible in the view.

```swift
    let store: StoreOf<DiscoverCategory>

    var body: some View {
        ScrollView { _ in
            WithPerceptionTracking {
                if store.hasError {
                    // Error sate placeholder
                } else if store.isEmpty {
                    // Empty sate placeholder
                } else {
                    content
                }
            }
            .padding(.vertical, Spacing.medium)
        }
    }

@MainActor
    private var content: some View {
        WithPerceptionTracking {
            LazyVStack(spacing: Spacing.huge) {
                // One of the following view is randomly not updated
                if store.showSubcategorySection {
                    SubcategoryListSectionView(store: store.scope(
                        state: \.subcategoryListSection,
                        action: \.subcategoryListSection
                    ))
                }
                ShowListSectionView(store: store.scope(
                    state: \.showListSection,
                    action: \.showListSection
                ))
                ProductListSectionView(store: store.scope(
                    state: \.productListSection,
                    action: \.productListSection
                ))
                ShowListGridSectionView(store: store.scope(
                    state: \.showList,
                    action: \.showList
                ))
            }
        }
    }
```

I have no runtime warning or nothing.

### Checklist

- [ ] I have determined whether this bug is also reproducible in a vanilla SwiftUI project.
- [ ] If possible, I've reproduced the issue using the `main` branch of this package.
- [X] This issue hasn't been addressed in an [existing GitHub issue](https://github.com/pointfreeco/swift-composable-architecture/issues) or [discussion](https://github.com/pointfreeco/swift-composable-architecture/discussions).

### The Composable Architecture version information

1.8.2

### Destination operating system

iOS 17

### Xcode version information

Xcode 15.2

### Swift Compiler version information

```shell
swift-driver version: 1.87.3 Apple Swift version 5.9.2 (swiftlang-5.9.2.2.56 clang-1500.1.0.2.5)
Target: arm64-apple-macosx14.0
```

**Comment by stephencelis** [comment]
@arnauddorgans `WithPerceptionTracking` needs to be used wherever a "lazy"/`@escaping` closure is, which `LazyVStack` is. Try moving the `WithPerceptionTracking` into the lazy view:

```diff
 @MainActor
 private var content: some View {
-  WithPerceptionTracking {
   LazyVStack(spacing: Spacing.huge) {
+    WithPerceptionTracking {
```

Because this isn't a bug in the library I'm going to convert to a discussion.

**Reply by arnauddorgans** [reply]
I thought `WithPerceptionTracking` was useful only before iOS 17, and bad usage of `WithPerceptionTracking` will result in a runtime warning 🤔

**Reply by arnauddorgans** [reply]
@stephencelis I still have issues when moving `WithPerceptionTracking` even if I think it is useless in iOS 17 to do this because Perception is not used.
I think it is an issue, not a discussion :/

**Reply by arnauddorgans** [reply]
I think this is related to https://github.com/pointfreeco/swift-composable-architecture/issues/2888

**Reply by stephencelis** [reply]
@arnauddorgans If you believe this is an issue please provide a full sample project that we can build and use. The code snippet above is just a view and doesn't compile without the supporting code.
